### PR TITLE
Add integration tests for AWS Lambda feature server

### PR DESCRIPTION
**Pull request by tsotnet**
Signed-off-by: Tsotne Tabidze <tsotne@tecton.ai>

**What this PR does / why we need it**: This PR does a couple of things:
* Build AWS Lambda feature server docker image for approved PRs
* Make PR integration test job dependent on the docker image build
* Merge ECR and Docker Hub image tagging logic (_get_version_for_aws, AWS_LAMBDA_FEATURE_SERVER_IMAGE) into one: _get_docker_image_version
  * Docker Hub image tag was constant `:aws`, which obviously couldn't support multiple releases (let alone integration tests)
  * The tag will now be either Feast public version (e.g. 0_14_0) for public releases, or git commit hash of HEAD (for integration tests).
  * For local development, I decided to use the latest released image up to the point where local code is at. For example, for local SDK version of `0.14.1.dev41+g1cbfa225.d20211104`, it'll use `0_14_1` version.
* Modify universal integration test configuration to deploy feature server for AWS provider. Also had to modify the registry to be on S3 instead of local, and had to make some changes how feature_store.yaml is written and how yaml file is parsed.
* Modify the universal integration test for online retrieval (test_online_retrieval) to check results from both SDK and AWS Lambda-based feature server

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by adchia** [comment]
going to hold off on review until the tests run successfully

File: sdk/python/tests/integration/feature_repos/repo_configuration.py
Comment (adchia) [inline sdk/python/tests/integration/feature_repos/repo_configuration.py:249]: This seems like it might break users who use Feast for testing their plugins?

File: sdk/python/feast/infra/aws.py
Comment (adchia) [inline sdk/python/feast/infra/aws.py:333]: Might be worth having a comment that that calls out that changes to this version for test should also impact the workflow yaml

File: sdk/python/tests/integration/online_store/test_universal_online.py
Comment (felixwang9817) [inline sdk/python/tests/integration/online_store/test_universal_online.py:232]: should we fail the test if the endpoint doesn't exist and the repo config has `feature_server=True`?

File: sdk/python/tests/integration/feature_repos/repo_configuration.py
Comment (felixwang9817) [reply sdk/python/tests/integration/feature_repos/repo_configuration.py:249]: I think it's probably fine for plugin tests to skip tests against serverless

File: .github/workflows/pr_integration_tests.yml
Comment (tsotnet) [reply .github/workflows/pr_integration_tests.yml]: I'll do this in a separate PR

File: sdk/python/feast/infra/feature_servers/aws_lambda/Dockerfile
Comment (adchia) [inline sdk/python/feast/infra/feature_servers/aws_lambda/Dockerfile:12]: is this change intended?

**Review by adchia — CHANGES_REQUESTED** [review]
lgtm pending fixing of flaky test

/lgtm

**Review by adchia — APPROVED** [review]

**Pull request merged**
